### PR TITLE
[vcpkg] Fixes and customizations to vcpkg_configure_make

### PR DESCRIPTION
**Pull request by ahmedyarub**
**Describe the pull request**
Changes were made to accommodate https://github.com/microsoft/vcpkg/pull/17577

- #### What does your PR fix?  
  Fixes variable name
  Adds the capability to choose the configuration script and disable shared/static flags. These changes are required for OpenSSL build.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Your answer`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: docs/maintainers/vcpkg_configure_make.md [hunk 1]
```diff
@@ -11,7 +11,9 @@ vcpkg_configure_make(
     [AUTOCONFIG]
     [USE_WRAPPERS]
     [DETERMINE_BUILD_TRIPLET]
+    [DISABLE_STATIC_SHARED]
     [BUILD_TRIPLET "--host=x64 --build=i686-unknown-pc"]
+    [CONFIG_FILE]
```
Comment (Neumann-A) [inline docs/maintainers/vcpkg_configure_make.md:16]: it should be `CONFIGURE_FILE` instead. Also please give further information why this is necessary

Comment (ahmedalnuaimi) [reply docs/maintainers/vcpkg_configure_make.md:16]: It's required for configuring libraries like OpenSSL where the default file name is Configure and not configure. OpenSSL port modernization is WIP currently.

Comment (ahmedyarub) [reply docs/maintainers/vcpkg_configure_make.md]: It's required for configuring libraries like OpenSSL where the default file name is Configure and not configure. OpenSSL port modernization is WIP currently.

**Comment by Neumann-A** [comment]
It's required for configuring libraries like OpenSSL where the default file name is Configure and not configure. OpenSSL port modernization is WIP currently.

Even if it is WIP open up a Draft PR and link it so that reviewers have context.

**Comment by Neumann-A** [comment]
Hmm seems a bit invalid for me since openssl `Configure` is using `#! /usr/bin/env perl` and not a bash script which `vcpkg_configure_make` assumes. I don't know how well it follows autotools rules in this case.

**Comment by ahmedyarub** [comment]
> Hmm seems a bit invalid for me since openssl `Configure` is using `#! /usr/bin/env perl` and not a bash script which `vcpkg_configure_make` assumes. I don't know how well it follows autotools rules in this case.

So do you propose running the configuration command directly without vcpkg_configure_make()?

**Comment by ahmedyarub** [comment]
> > It's required for configuring libraries like OpenSSL where the default file name is Configure and not configure. OpenSSL port modernization is WIP currently.
> 
> Even if it is WIP open up a Draft PR and link it so that reviewers have context.

Added to the description.

**Comment by JackBoosY** [comment]
You should modify the comments in `vcpkg_configure_make.cmake` and regenerate the documentation.

**Comment by ahmedyarub** [comment]
> You should modify the comments in `vcpkg_configure_make.cmake` and regenerate the documentation.

Done

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 2]
```diff
@@ -489,14 +502,16 @@ function(vcpkg_configure_make)
...
-    if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
-        list(APPEND _csc_OPTIONS --enable-shared --disable-static)
-    else()
-        list(APPEND _csc_OPTIONS --disable-shared --enable-static)
+    if(NOT _csc_DISABLE_STATIC_SHARED)
+        if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+            list(APPEND _csc_OPTIONS --enable-shared --disable-static)
+        else()
+            list(APPEND _csc_OPTIONS --disable-shared --enable-static)
+        endif()
```
Comment (JackBoosY) [inline scripts/cmake/vcpkg_configure_make.cmake:514]: I think we should use `VCPKG_LIBRARY_LINKAGE` to determine which type should be used, that's suitable for all triplets.
Do you use it in any scene?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: scripts/cmake/vcpkg_configure_make.cmake (hunk 2 above)
Comment (JackBoosY) [reply scripts/cmake/vcpkg_configure_make.cmake:514]: For those ports that need to be forced to use static construction, you should use the following code:
```
if (VCPKG_HOST_IS_WINDOWS/VCPKG_HOST_IS_LINUX/VCPKG_HOST_IS_OSX)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()
```

Comment (ahmedyarub) [reply scripts/cmake/vcpkg_configure_make.cmake:514]: @JackBoosY as I mentioned in the description, this was made to accommodate https://github.com/microsoft/vcpkg/pull/17577
The configuration script there does not recognize these parameters. It has its own parameters for choosing between static and dynamic library build.

Comment (ahmedyarub) [reply scripts/cmake/vcpkg_configure_make.cmake:514]: In OpenSSL I'll be disabling these flags and adding logic specific to its build scripts.

Comment (JackBoosY) [reply scripts/cmake/vcpkg_configure_make.cmake:514]: As I said above, I think the best way is add the code I provided.
@Neumann-A What do you think about?

**Comment by ahmedyarub** [comment]
Closed because depending PR is not valid anymore.

**Pull request closed without merging**